yjit/src/core.rs

```diff
@@ -286,63 +287,85 @@ pub enum TypeDiff {
     Incompatible,
 }
 
-// Potential mapping of a value on the temporary stack to
-// self, a local variable or constant so that we can track its type
 #[derive(Copy, Clone, Eq, Hash, PartialEq, Debug)]
-pub enum TempMapping {
-    MapToStack, // Normal stack value
-    MapToSelf,  // Temp maps to the self operand
-    MapToLocal(LocalIndex), // Temp maps to a local variable with index
-                //ConstMapping,         // Small constant (0, 1, 2, Qnil, Qfalse, Qtrue)
+pub enum TempMappingKind
+{
+    MapToStack = 0,
+    MapToSelf = 1,
+    MapToLocal = 2,
 }
 
-// Index used by MapToLocal. Using this instead of u8 makes TempMapping 1 byte.
+// Potential mapping of a value on the temporary stack to
+// self, a local variable or constant so that we can track its type
```

Comment:
It'd be useful to summarize the structure of `u8` somewhere, maybe here. It currently requires you to read multiple functions to understand how it works.

```suggestion
// self, a local variable or constant so that we can track its type
//
// The highest two bits represent TempMappingKind, and the rest of
// the bits are used differently across different kinds.
// * MapToStack: The lowest 5 bits are used for mapping `Kind`.
// * MapToSelf: Not used; the type is stored in self_type.
// * MapToLocal: The lowest 3 bits store the index of a local variable.
```